File: app/src/main/java/me/zipi/navitotesla/db/ConditionEntity.java
package me.zipi.navitotesla.db;

import java.util.Date;

import androidx.room.Entity;
import androidx.room.Ignore;
import androidx.room.Index;
import androidx.room.PrimaryKey;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity(tableName = "condition", indices = @Index(value = {"type", "name"}, unique = true))
@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor(onConstructor = @__({@Ignore}))
public class ConditionEntity {

    @PrimaryKey(autoGenerate = true)
    private Integer id;

    private String name;
    // wifi bluetooth
    private String type;

    private Date created;

}

File: app/src/main/kotlin/me/zipi/navitotesla/db/ConditionEntity.kt
package me.zipi.navitotesla.db

import androidx.room.Entity
import androidx.room.Index
import androidx.room.PrimaryKey
import java.util.Date

@Entity(tableName = "condition", indices = [Index(value = ["type", "name"], unique = true)])
class ConditionEntity(
    @PrimaryKey(autoGenerate = true)
    val id: Int? = null,
    val name: String,

    // wifi bluetooth
    val type: String,
    val created: Date,
)
